quickstarts/microsoft.documentdb/cosmosdb-sql-rbac/main.bicep

```diff
@@ -0,0 +1,68 @@
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Cosmos DB account name, max length 44 characters')
+param accountName string = 'sql-rbac-${uniqueString(resourceGroup().id)}'
+
+@description('Friendly name for the SQL Role Definition')
+param roleDefinitionName string = 'My Read Write Role'
+
+@description('Data actions permitted by the Role Definition')
+param dataActions array = [
+  'Microsoft.DocumentDB/databaseAccounts/readMetadata'
+  'Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/*'
+]
+
+@description('Object ID of the AAD identity. Must be a GUID.')
+param principalId string
+
+var account_name = toLower(accountName)
+var locations = [
+  {
+    locationName: location
+    failoverPriority: 0
+    isZoneRedundant: false
+  }
+]
+var roleDefinitionId = guid('sql-role-definition-', accountName_resource.id)
+var roleAssignmentId = guid('sql-role-assignment-', accountName_resource.id)
```

Comment:
this must be based on the principal, role and scope, e.g.

guid(databaseAccount.id, roleDefinitionId, principalId)